prompt.go

```diff
@@ -11,6 +11,10 @@ import (
 // Executor is called when user input something text.
 type Executor func(string)
 
+// Exitor is called after user input something text, to check if
+// prompt must stop and exit
+type Exitor func(string) bool
```

Comment:
I think most users mind the timing of this function call. So I prefer a following godoc comment. May I have your opinion?

`Exitor returns true if the prompt must stop and exit. This function is called before executing an Executor.`

Reply:
@c-bata You are right, I missed the executor part.
To avoid any surprise for the user, I would like to exit *after* running the executor.
But that means p.in.TearDown() is executed before the return.

So I am back to my `inIsTearedDown` boolean (I cannot call exitor twice in the deferred function, because that function does not have access to e.input)